### DISCUSSION TITLE
Varying inertial properties but constant kinematics

### DISCUSSION THREAD

**Discussion by MaximilianDio** [stack-of-tasks/pinocchio#1874] [Q&A] [answered]
Hi there. 

I would like to optimize dynamic properties like mass, center of gravity, inertia etc., however, the kinematics and therefore the joint frames do not change. Is there a way to update the model inertial properties without building the complete model, running kinematics etc.

kind regards 
Max

**Comment by jcarpent** [comment, marked as answer]
You can change the inertia property of the model manually.
For instance, `model.inerias[joint_id].mass = 2.` will set the Inertia quantity supported by `joint_id` to the value 2.

**Reply by MaximilianDio** [reply]
Ok that is helpful but also the center of  mass? 

Lets say I have a joint, and a mass is attached to it at an offset of r = [rx,ry,rz] in the joints coordinate frame, I could adjust the inertia matrix using Huygen-Steiner but I would still need to change the COM

**Reply by MaximilianDio** [reply]
Ah ok I see I can alter all the quantities that is perfect!

**Reply by jcarpent** [reply]
Yes, every quantity can be changed.
